uxarray/grid.py

```diff
@@ -84,18 +81,11 @@ def __init__(self, *args, **kwargs):
 
         # check if initializing from string
         # TODO: re-add gridspec initialization when implemented
-        elif isinstance(data_arg, str):
-            # check if file exists
-            if not os.path.isfile(data_arg):
-                raise RuntimeError("File not found: " + data_arg)
-
-            self.filepath = data_arg
-            # call the appropriate reader
-            self.__from_file__()
-
-        # check if invalid initialization
+        elif isinstance(data_arg, xr.Dataset):
+            self.xr_ds = data_arg
```

Comment:
I am concerned about keeping this as an instance attribute because it introduces redundant memory use taht'd be an issue with large datasets and we'd not actually need this original dataset any longer after we construct the UGRID-aware one.

Reply:
If you assign to `self.ds` then it'll just be overwritten after the "decode" step (e.g. `read_script`)?

Reply:
I think that'd make the code confusing and less readable (`self.ds` is the UGRID-aware ds after all the decoding, but it'd seem like initiated with the original ds coming from the data ingest. One would need to track all the way down to see `self.ds` is indeed the ugrid-aware one)

Reply:
OK good point; could just pass `dataset` (or `data_arg`) as arg to `self._from_ds__`